src/open_inwoner/openzaak/import_export.py

```diff
@@ -0,0 +1,146 @@
+from collections import defaultdict
+from typing import Any, Generator, Self
+
+from django.core import serializers
+from django.core.files.storage import Storage
+from django.db.models import QuerySet
+
+from .models import (
+    CatalogusConfig,
+    ZaakTypeConfig,
+    ZaakTypeInformatieObjectTypeConfig,
+    ZaakTypeResultaatTypeConfig,
+    ZaakTypeStatusTypeConfig,
+)
+
+logger = logging.getLogger(__name__)
+
+
+@dataclasses.dataclass
+class CatalogusConfigExport:
+    """Gather and export CatalogusConfig(s) and all associated relations."""
+
+    catalogus_configs: QuerySet
+    zaak_types: QuerySet
+    informatie_object_types: QuerySet
+    status_types: QuerySet
+    resultaat_types: QuerySet
+
+    def __iter__(self) -> Generator[QuerySet, Any, None]:
+        yield from (
+            self.catalogus_configs,
+            self.zaak_types,
+            self.informatie_object_types,
+            self.status_types,
+            self.resultaat_types,
+        )
+
+    def __eq__(self, other) -> bool:
+        for a, b in zip(self, other):
+            if a.difference(b).exists():
+                return False
+
+        return True
+
+    @classmethod
+    def from_catalogus_configs(cls, catalogus_configs: QuerySet) -> Self:
+        if not isinstance(catalogus_configs, QuerySet):
+            raise ValueError(
+                f"`catalogus_configs` is not a QuerySet, but a {type(catalogus_configs)}"
+            )
+
+        if catalogus_configs.model != CatalogusConfig:
+            raise ValueError(
+                f"`catalogus_configs` is of type {catalogus_configs.model}, not CatalogusConfig"
+            )
+
+        zaak_types = ZaakTypeConfig.objects.filter(catalogus__in=catalogus_configs)
+        informatie_object_types = ZaakTypeInformatieObjectTypeConfig.objects.filter(
+            zaaktype_config__in=zaak_types
+        )
+        status_types = ZaakTypeStatusTypeConfig.objects.filter(
+            zaaktype_config__in=zaak_types
+        )
+        resultaat_types = ZaakTypeResultaatTypeConfig.objects.filter(
+            zaaktype_config__in=zaak_types
+        )
+
+        return cls(
+            catalogus_configs=catalogus_configs,
+            zaak_types=zaak_types,
+            informatie_object_types=informatie_object_types,
+            status_types=status_types,
+            resultaat_types=resultaat_types,
+        )
+
+    def as_dicts_iter(self) -> Generator[dict, Any, None]:
+        for qs in self:
+            serialized_data = serializers.serialize(
+                queryset=qs,
+                format="json",
+                use_natural_foreign_keys=True,
+                use_natural_primary_keys=True,
+            )
+            json_data: list[dict] = json.loads(
+                serialized_data,
+            )
+            yield from json_data
+
+    def as_jsonl_iter(self) -> Generator[str, Any, None]:
+        for row in self.as_dicts():
+            yield json.dumps(row)
+            yield "\n"
```

Comment:
You  could remove the newline from here and move it into `as_jsonl` below (i.e. `"\n".join(self.as_json_iter())`. This would make the function more general (could then be named "as_json_iter") and separate neatly between data and presentation